### PR TITLE
Removed jaggery.js component from publisher portal and replaced it with jsp. Migrated the publisher portal as a webapp.

### PR DESCRIPTION
**Pull request by DinithHerath**

### CODE REVIEW

File: portals/publisher/src/main/webapp/services/login/idp.jsp [hunk 1]
```diff
@@ -0,0 +1,188 @@
+<%--
+  ~ Copyright (c) 2017, WSO2 LLC (http://www.wso2.org) All Rights Reserved.
```
Comment (tharikaGitHub) [inline portals/publisher/src/main/webapp/services/login/idp.jsp:2]: We have LLC licence in some files while some others have Inc. Should we make it consistent?

**Review by tharikaGitHub — APPROVED** [review]

File: portals/publisher/src/main/webapp/services/login/idp.jsp (hunk 1 above)
Comment (DinithHerath) [reply portals/publisher/src/main/webapp/services/login/idp.jsp:2]: +1 sorry for that….

File: portals/publisher/src/main/java/org/wso2/carbon/apimgt/ui/publisher/Util.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
+/*
+
+
+ *  Copyright (c) 2017, WSO2 LLC (http://www.wso2.org) All Rights Reserved.
```
Comment (dushaniw) [inline portals/publisher/src/main/java/org/wso2/carbon/apimgt/ui/publisher/Util.java:4]: @DinithHerath there is formatting issue in the license header. Shall we fix this and the year as 2017 - 2023

File: portals/publisher/src/main/java/org/wso2/carbon/apimgt/ui/publisher/Util.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        if (!nestedJson.containsKey(pathStrings[pathStrings.length - 1])) {
+            return null;
+        }
+        return nestedJson.get(pathStrings[pathStrings.length - 1]);
+    }
+    
+    /**
+     * Get the loopback (localhost) origin (scheme + hostname + port), This origin is used for making
+     * internal(within the web app node), API calls.For example DCR call, Token generation, User Info, Token Introspect,
+ Revoke etc.
```
Comment (dushaniw) [inline portals/publisher/src/main/java/org/wso2/carbon/apimgt/ui/publisher/Util.java:84]: formatting issue

File: portals/publisher/src/main/webapp/WEB-INF/web.xml [hunk 4]
```diff
@@ -0,0 +1,205 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    ~ Copyright (c) 2017, WSO2 LLC (http://www.wso2.org) All Rights Reserved.
```
Comment (dushaniw) [inline portals/publisher/src/main/webapp/WEB-INF/web.xml:3]: shall we add the license header year as  `2017-2023`

File: portals/publisher/src/main/webapp/services/login/idp.jsp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  ~
```
Comment (dushaniw) [inline portals/publisher/src/main/webapp/services/login/idp.jsp:3]: lets fix the year in hear as well

File: portals/publisher/src/main/webapp/services/login/idp.jsp [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+<%@page import="org.wso2.carbon.apimgt.impl.IDPConfiguration"%>
+<%@page import="org.wso2.carbon.apimgt.impl.utils.APIUtil"%>
+<%@page import="org.wso2.carbon.apimgt.ui.publisher.Util"%>
+<%@page import="java.util.Map"%>
+
+<%@include file="../constants.jsp" %>
+
+<%@page trimDirectiveWhitespaces="true" %>
+
+<%    Log log = LogFactory.getLog(this.getClass());
```
Comment (dushaniw) [inline portals/publisher/src/main/webapp/services/login/idp.jsp:45]: formatting issue

File: portals/publisher/src/main/webapp/services/login/introspect.jsp [hunk 7]
```diff
@@ -0,0 +1,85 @@
+<%--
+  ~ Copyright (c) 2017, WSO2 LLC (http://www.wso2.org) All Rights Reserved.
```
Comment (dushaniw) [inline portals/publisher/src/main/webapp/services/login/introspect.jsp:2]: 2017-2023

File: portals/publisher/src/main/webapp/services/login/idp.jsp (hunk 6 above)
Comment (dushaniw) [reply portals/publisher/src/main/webapp/services/login/idp.jsp:45]: lets fix all the places

File: portals/publisher/src/main/webapp/services/login/idp.jsp (hunk 5 above)
Comment (dushaniw) [reply portals/publisher/src/main/webapp/services/login/idp.jsp:3]: lets fix all the places

File: portals/publisher/src/main/java/org/wso2/carbon/apimgt/ui/publisher/Util.java [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+     * @return String
+     */
+    public static String getAppContextForServerUrl(String context, String proxyContext) {
+        String appContext = context;
+        if (proxyContext != null && !proxyContext.isEmpty()) {
+            appContext = appContext.replace(proxyContext, "");
+        }
+        return appContext;
+    }
+}
```
Comment (HeshanSudarshana) [inline portals/publisher/src/main/java/org/wso2/carbon/apimgt/ui/publisher/Util.java:183]: Add a new line to the end. Change this in other files as well.

File: portals/publisher/src/main/java/org/wso2/carbon/apimgt/ui/publisher/Util.java [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+/**
+ * Class for the utility functions needed by the services
+ */
+public class Util {
+
+    /**
+     * Read a json file from the directory and output as a Map object.
+     *
+     * @param path    required parameter
```
Comment (HeshanSudarshana) [inline portals/publisher/src/main/java/org/wso2/carbon/apimgt/ui/publisher/Util.java:45]: Add descriptive details to the params of the method documentation. Add only the description for the return tag not the return type. Change in other places as well.

**Pull request merged**